pkg/kubecli/kubevirt_test_utils.go

```diff
@@ -0,0 +1,54 @@
+package kubecli
+
+import (
+	"errors"
+
+	k8smetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+)
+
+// GetMockKubevirtClientFromClientConfig, MockKubevirtClientInstance and InvalidMockClient are used to create a mechanism
+// for overriding the actual kubevirt client access. This is useful when the unit tested code invokes GetKubevirtClientFromClientConfig()
+// and therefore the unit test code cannot generate the mock client directly. In such a case following steps are needed:
+// (1) set InvalidMockClient to true/false to control the behavior of GetKubevirtClientFromClientConfig()
+// (2) if set to "false", you should override the GetKubevirtClientFromClientConfig() closure with GetMockKubevirtClientFromClientConfig()
+// (3) then create the instance of the client, and assign into MockKubevirtClientInstance before the tests start:
+//     ctrl := gomock.NewController(GinkgoT())
+//     kubecli.MockKubevirtClientInstance = kubecli.NewMockKubevirtClient(ctrl)
+// (4) rest of the kubevirt mocking is automatically generated in generated_mock_kubevirt.go
+
+// InvalidMockClient is a parameter to control behavior of mock objects
+var InvalidMockClient bool
+
+// MockKubevirtClientInstance is a reference to the kubevirt client that could be manipulated by the test code
+var MockKubevirtClientInstance *MockKubevirtClient
+
+// GetMockKubevirtClientFromClientConfig is an entry point for testing, could be used to override GetKubevirtClientFromClientConfig
+func GetMockKubevirtClientFromClientConfig(cmdConfig clientcmd.ClientConfig) (KubevirtClient, error) {
```

Comment:
Could you just create another function for the error case?

Reply:
ok. yes, better than the global var

Reply:
fixed